scripts/staging/fm_binclass.dml

```diff
@@ -19,32 +19,89 @@
 #
 #-------------------------------------------------------------
 
-/*
+
+/**
  * Factorization Machines for binary classification.
  */
 
 # Imports
 source("staging/fm.dml") as fm
 source("nn/layers/sigmoid.dml") as sigmoid
 source("nn/layers/log_loss.dml") as log_loss
+source("nn/optim/adam.dml") as adam
 
 train = function(matrix[double] X)
-    return (matrix[double] out) {
+    return (matrix[double] Y, matrix[double] X_val, matrix[double] Y_val) {
```

Comment:
Ah, no have `train` accept `X`, `Y`, `X_val`, and `Y_val` as input parameters, and return the model weights `w0`, `W`, `V`.